com.unity.render-pipelines.core/Runtime/XR/XROcclusionMesh.cs

```diff
@@ -0,0 +1,173 @@
+namespace UnityEngine.Rendering
+{
+    // Helper class to render occlusion meshes.
+    // If possible, the mesh for each view will be combined into one mesh to reduce draw calls.
+    internal class XROcclusionMesh
+    {
+        XRPass m_Pass;
+        Mesh m_CombinedMesh;
+        Material m_Material;
+        int m_CombinedMeshHashCode;
+
+        static readonly ProfilingSampler k_OcclusionMeshProfilingSampler = new ProfilingSampler("XR Occlusion Mesh");
+
+        internal XROcclusionMesh(XRPass xrPass)
+        {
+            m_Pass = xrPass;
+        }
+
+        internal void SetMaterial(Material mat)
+        {
+            m_Material = mat;
```

Comment:
I wonder if this created a copy of the material instead of holding a reference. shouldnt we use ref keyword?

Reply:
Classes in C# (like Material) are reference type by default so no need for ref keyword here.